### PR TITLE
[Propalsal] propalsal to add edge-autonomy-e2e-tests

### PR DESCRIPTION
**Pull request by lorrielau**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Add edge-autonomy-e2e-tests for Openyurt.

#### Special notes for your reviewer:

@Congrool 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Congrool** [comment]
> E2e-tests timed out. How can I re-run the checks? :>

Currently, we do not have something like `/retest`😢. You can `git commit --amend --no-edit --signoff` and then push to your repo again, e2e-test then will be re-triggered.

File: docs/proposals/20220901-add-edge-autonomy-e2e-tests.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- Adding e2e-tests for modules
+  - Kubelet
+    - Run a busybox pod(which does not need cache) on each edge node when they're still online;
+    - Check if the pod has successfully restarted after disconnection from cloud node. If so, kubelet is working normally.
+  - Yurthub
+    - After disconnection, use the cat/curl command on 127.0.0.1:10267/v1/healthz to see if yurthub  successfully restarts and is in healthy running status.
+  - Kube-proxy
+    - After disconnection, kill the kube-proxy pod, and request to ping other nodes. If the requests succeed before timeout, kube-proxy works. This requires there are services exposing instances on each edge node before the requests are made.
+  - Flannel
+    - After disconnection, ping other edge nodes using cluster ip. If it works, flannel had restarted and functioned successfully.
```
Comment (Congrool) [inline docs/proposals/20220901-add-edge-autonomy-e2e-tests.md:75]: We should first be in a pod and exec command to ping another pod on another node with PodIP.
Additionally, before test, we have to restart flannel and these pods on both nodes. The restart sequence is matter, we should first restart flannel and then these test pods. It helps us to test IPAM function of cni.

File: docs/proposals/20220901-add-edge-autonomy-e2e-tests.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - CoreDns
+    - After disconnection, using dig-command to validate if coredns is capable of resolution and is working.
```
Comment (Congrool) [inline docs/proposals/20220901-add-edge-autonomy-e2e-tests.md:77]: Also dig after restarting coredns, and we should specify the dns server of dig to be coreDNS. And the domain name we dig should be in the cluster, such as service name.

**Comment by Congrool** [comment]
Could you please also describe a little more about how do we do autonomy test, including 
1. how could we know if test succeeds on the host, since these tests are running in the kind node which is a container?
2. how to distinguish original test and autonomy test, and how to run them respectively?
3. which way do these tests run in, in parallel, in serial or in any other special order?

And also add details about how to run each autonomy test on kubelet, kube-proxy, flannel, coreDNS and yurthub. We can describe them in steps.

**Comment by lorrielau** [comment]
Done, please take a look.

**Comment by lorrielau** [comment]
Added 2 picture.

**Comment by lorrielau** [comment]
Modified

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**